src/sentry/static/sentry/app/components/listViewContainer.jsx

```diff
@@ -0,0 +1,131 @@
+      search = query.search ? query.search : search;
+      cursor = query.cursor ? query.cursor : cursor;
+      groupBy = query.groupBy ? query.groupBy : groupBy;
+    }
+
+    this.onSearch(search, groupBy, cursor);
+  }
+
+  onSearch(search, groupBy, cursor) {
+    // Corresponding in substance search:
+    // const isGroupHeader = groupBy !== 'substance';
+    // this.props.substanceSearchEntriesGet(search, groupBy, cursor, isGroupHeader);
+
+    this.props.getEntries(search, groupBy, cursor); // TODO: not implemented for workbatch
+
+    // Add search to history
+    const location = this.props.location;
+    const query = {
+      ...location.query,
+      search,
+      groupBy,
+      cursor,
+    };
+
+    // TODO: Might want to do this at the caller's (with an event) so we don't mess up the URL
+    // if the caller doesn't want it. Or have a prop setting for this.
+    browserHistory.push({
+      pathname: location.pathname,
+      query,
+    });
+  }
+
+  onGroup(groupBy) {
+    this.setState({groupBy: {value: groupBy}});
+    const {search} = this.props;
+    const {cursor} = this.props.pagination;
+    this.onSearch(search, groupBy, cursor);
+  }
+
+  onSort(e) {}
+
+  onCursor(cursor) {
+    const {search, groupBy} = this.props;
+    this.onSearch(search, groupBy, cursor);
+  }
+
+  onSavedSearchCreate() {
+    // TODO
+  }
+
+  toggleAll() {
+    this.props.toggleSelectAll(null);
+  }
+
+  render() {
+    return (
+      <div className="stream-row">
+        <div className="stream-content">
+          <ListFilters
+            access={this.props.access}
+            onSavedSearchCreate={this.onSavedSearchCreate}
+            searchPlaceholder={this.props.searchHelpText}
+            groupOptions={this.props.groupOptions}
+            grouping={this.props.listViewState.groupBy}
+            onGroup={this.onGroup}
+            onSearch={this.onSearch}
+            orgId={this.props.organization.id}
+            query={this.props.listViewState.search} // Rename to search
+          />
+          <ListView
+            orgId={this.props.organization.id}
+            columns={this.props.columns()}
+            dataById={this.props.byIds}
+            visibleIds={this.props.listViewState.visibleIds}
+            selectedIds={this.props.listViewState.selectedIds}
+            loading={this.props.loading}
+            canSelect={true}
+            allVisibleSelected={this.props.listViewState.allVisibleSelected}
+            toggleAll={this.toggleAll}
+            toggleSingle={this.props.toggleSingle}
+            listActionBar={null}
+          />
+        </div>
+      </div>
+    );
+
+    // TODO: Enable pagination
+    // {this.props.pagination.paginationEnabled &&
+    //   this.props.pagination.pageLinks && (
+    //     <Pagination
+    //       pageLinks={this.props.pagination.pageLinks}
+    //       onCursor={this.onCursor}
+    //     />
+    //   )}
+  }
+}
+
+ListViewContainer.propTypes = {
+  ...ClimsTypes.List,
```

Comment:
Is it useful to centralize ClimsType.List? This seem to be a generic list view component. Isn't it a good change this will be the only place where it's used?

Reply:
This is used in several places in a coming PR where it defines the structure of the redux stores.

Reply:
ok